### PR TITLE
feat: merge all release-alpha 3.0.0 features to master

### PR DESCRIPTION
**Pull request by christjt**

### CODE REVIEW

File: viewer/packages/modeldata-api/src/applyDefaultModelTransformation.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ export function applyDefaultModelTransformation(matrix: THREE.Matrix4, format: F
       matrix.premultiply(cadFromCdfToThreeMatrix);
       break;
 
+    case File3dFormat.RevealCadModel:
```
Comment (haakonflatval-cognite) [inline viewer/packages/modeldata-api/src/applyDefaultModelTransformation.ts:18]: This is a duplicate of the above case.

File: viewer/packages/cad-geometry-loaders/src/sector/culling/ByVisibilityGpuSectorCuller.ts [hunk 2]
```diff
@@ -161,32 +70,18 @@ export class ByVisibilityGpuSectorCuller implements SectorCuller {
...
-    const forcedDetailedSectorCount = nonDiscarded.filter(x => !Number.isFinite(x.priority)).length;
-    const accumulatedPriority = nonDiscarded
-      .filter(x => Number.isFinite(x.priority) && x.priority > 0)
-      .reduce((sum, x) => sum + x.priority, 0);
-    const takenDetailedPercent = ((100.0 * (takenSectorCount - takenSimpleCount)) / totalSectorCount).toPrecision(3);
-    const takenPercent = ((100.0 * takenSectorCount) / totalSectorCount).toPrecision(3);
+    const spentBudget = takenSectors.computeSpentBudget();
+
+    const takenDetailedPercent = (
+      (100.0 * (spentBudget.loadedSectorCount - spentBudget.loadedSectorCount)) /
```
Comment (haakonflatval-cognite) [inline viewer/packages/cad-geometry-loaders/src/sector/culling/ByVisibilityGpuSectorCuller.ts:76]: `spentBudget.loadedSectorCount - spentBudget.loadedSectorCount` is always 0. Not entirely sure what it should be instead here, though :thinking:

**Review by haakonflatval-cognite — COMMENTED** [review]
Just to clarify the warnings given by LGTM.com

File: viewer/packages/cad-geometry-loaders/src/sector/culling/ByVisibilityGpuSectorCuller.ts (hunk 2 above)
Comment (christjt) [reply viewer/packages/cad-geometry-loaders/src/sector/culling/ByVisibilityGpuSectorCuller.ts:76]: lol, @larsmoa any input here?

**Review by haakonflatval-cognite — APPROVED** [review]
Those sure are some nice 40,092 additions and 17,114 deletions :+1:

**Pull request merged**